### PR TITLE
MOTECH-2839: Added Cohort Query Member Task Trigger

### PR DESCRIPTION
**Pull request by afijal**

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by kmadej** [comment]
test this please

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package org.motechproject.openmrs.tasks.builder;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Helper class for building display names for the Commcare task channel and data provider.
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java]: fix javadoc

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ */
+public final class DisplayNameHelper {
+
+    /**
+     * Builds a display name in the form of "subject: name [config-name]". Example - Received Form: Birth [myConfig].
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java:11]: fix javadoc

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+     * If the name is blank, this will be formatted as "name [config-name]".
+     * @param subject the subject with which the display name should start
+     * @param name the name of the element (form, case, etc.), if blank will be omitted
+     * @param applicationName the name of the application
+     * @param configName the name of the configuration
+     * @return the display name
+     */
+    public static String buildDisplayName(String subject, String name, String applicationName, String configName) {
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java]: applicationName ? What is it ?

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        if (StringUtils.isBlank(name) && StringUtils.isBlank(applicationName)) {
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/DisplayNameHelper.java]: use one variable and return at the end of method

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/TriggerBuilder.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import org.motechproject.openmrs.service.OpenMRSConfigService;
+import org.motechproject.openmrs.tasks.constants.DisplayNames;
+import org.motechproject.openmrs.tasks.constants.Keys;
+import org.motechproject.tasks.contract.EventParameterRequest;
+import org.motechproject.tasks.contract.TriggerEventRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/TriggerBuilder.java]: javadoc

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/TriggerBuilder.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+ * Created by user on 12.08.16.
+ */
+public class TriggerBuilder {
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/builder/TriggerBuilder.java]: change the name to OpenMRSTriggerBuilder

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/DisplayNames.java [hunk 7]
```diff
@@ -6,6 +6,7 @@
 public final class DisplayNames {
 
     public static final String CONFIG_NAME = "openMRS.configuration.name";
+    public static final String COHORT_GOTMEMBER = "Got member from cohortquery";
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/constants/DisplayNames.java]: sth wrong here

**Comment by kmadej** [comment]
something is wrong with this PR, please adjust git tree

**Pull request closed without merging**